### DISCUSSION TITLE
诚心发问，有关v2ray 安全性的小疑问

### DISCUSSION THREAD

**Discussion by eddieran** [v2fly/v2ray-core#1459] [Q&A] [answered]
用v2ray也一两年了，近期可能想接触一些加密货币的事情，不知道有大佬能帮忙解惑一下安全性的疑虑吗？
版本: v4.44.0
 底层传输安全 (TLS) = tls
 传输协议 (Network) = ws
 伪装类型 (header type) = none

这样的配置，站在vps提供商的角度上，能看出来你在做什么吗？
感觉tls理论上应该是安全的

大家有过类似的疑虑吗？
毕竟crypto的钱包(比如metamask)的访问，很多dApp的访问，都需要挂梯子

**Comment by kallydev** [comment, marked as answer]
这与如何配置 v2ray 无关，只取决于你的服务器最终访问的第三方服务是否提供 HTTPS 等传输方式。即使访问被代理的网站时使用了 HTTPS，但 SNI 仍然是明文，服务器提供商仍然可以知道你访问了哪些网站，但并不知道具体内容（除非你关闭了 TLS 证书验证）。

可以通过 Tor 网络模型进行类比，Tor 默认提供三层路由，第一层只知道加密数据和来源，第二层只知道加密数据，第三层则知道部分明文（HTTPS 除外）数据内容和数据去向，在 Tor 模型中如果第一层或第三层由恶意服务器伪装，很有可能带来风险。同样，在单节点代理网络中，如果节点提供商不可信，相当于 Tor 网络中第一层和第三层直接串通，安全性微乎其微。

Update:
忘记回答加密货币相关内容，加密货币一般都是使用的公私钥，例如以太坊就是使用的 ECDSA（椭圆曲线数字签名算法），这也是 TLS 所使用的技术之一。在加密货币钱包中，私钥可以完全不接触网络。Metamask 同时提供热钱包（助记词）和冷钱包两种功能，如果选择热钱包，Metamask 会替你保管私钥。建议自行生成私钥并保管妥当，因为私钥一旦泄漏或丢失你便失去了钱包的完整所有权。

**Reply by eddieran** [reply]
多谢解惑！
大概了解了！其实走了HTTPS访问正规证书网站，即便服务器提供商能够看到你的SNI，但其实并不能看到具体内容。
对于metamask来说，安全性排名大概是
冷钱包 > 热钱包 + 自行保管私钥 > 热钱包metamask为你管理私钥 
我已经准备购买硬件钱包，还是硬件钱包才是最安心的

**Reply by Kylejustknows** [reply]
> 多谢解惑！ 大概了解了！其实走了HTTPS访问正规证书网站，即便服务器提供商能够看到你的SNI，但其实并不能看到具体内容。 对于metamask来说，安全性排名大概是 冷钱包 > 热钱包 + 自行保管私钥 > 热钱包metamask为你管理私钥 我已经准备购买硬件钱包，还是硬件钱包才是最安心的

你多虑了。 不管冷热钱包，只要助记密码写纸上，本地钱包文件加强密码，都已经足够安全。硬件钱包没有带来理论安全收益反而可能增加了另一层风险和麻烦。写在纸上，存入银行保险库，登记好您去世后继承人等信息（不要低估一个普通人，每年因为各种原因突然死亡的概率）

你唯一的风险，来自于你的手机和电脑是否干净。
现在越来越多的木马，专用于盗取钱包。 
如果是瞬间中毒，会偷走你的本地钱包文件（如果没有本地强密码你就瞬间GG了）
如果是持续中毒，会长期记录你所有密码，会篡改钱包和网页的界面（你发给朋友的币，却发给了黑客），什么冷钱包热钱包硬件钱包都没用。

最简单的整体解决方案：助记词写纸上存入银行保险箱，使用一台品牌手机长期作为唯一的钱包客户端。

（手机最好使用海外版固件，国外一直有技术人员检查手机厂家自己有没有自己偷数据，而国内对隐私不敏感有些乱。自己装的其它APP不越狱也无法获取钱包数据，所以手机可以作为日常使用机）
